File: typescript/problems/1.两数之和.ts
/*
 * @lc app=leetcode.cn id=1 lang=typescript
 *
 * [1] 两数之和
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
  const map = new Map()
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i],
      l = target - num
    if (map.has(l)) return [map.get(l), i]
    map.set(num, i)
  }
  return []
}
// @lc code=end

File: leetcode/leetcode-hot-100/python3/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = dict()
        for i, num in enumerate(nums):
            l = target - num
            if l in dic:
                return [dic[l], i]
            dic[num] = i
        return []


# @lc code=end
